runtime/vm/vmthread.c

```diff
@@ -399,6 +399,11 @@ void threadCleanup(J9VMThread * vmThread, UDATA forkedByVM)
 	/* We are dead at this point. Clear the suspend bit prior to triggering the thread end hook */
 	clearHaltFlag(vmThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
 
+#if (JAVA_SPEC_VERSION >= 14)
+    /* Refresh java.lang.Thread interrupt value so it is acessible when vm thread is cleaned up. */
+    J9VMJAVALANGTHREAD_SET_DEADINTERRUPT(vmThread, vmThread->threadObject, omrthread_interrupted(vmThread->osThread));
```

Comment:
We can't modify an object without vm access.  This will need to be done before the call to `releaseVMAccess`